rofl-appd/src/routes/tx.rs

```diff
@@ -0,0 +1,141 @@
+use std::sync::Arc;
+
+use rocket::{http::Status, serde::json::Json, State};
+use serde_with::serde_as;
+
+use oasis_runtime_sdk::{modules::rofl::app::client::SubmitTxOpts, types::transaction};
+use oasis_runtime_sdk_evm as evm;
+
+use crate::state::Env;
+
+/// Transaction.
+#[serde_as]
+#[derive(Clone, Debug, serde::Deserialize)]
+#[serde(tag = "kind", content = "data")]
+pub enum Transaction {
+    /// Standard Oasis SDK transaction.
+    #[serde(rename = "std")]
+    Std(#[serde_as(as = "serde_with::hex::Hex")] Vec<u8>),
+
+    /// Ethereum transaction.
+    #[serde(rename = "eth")]
+    Eth {
+        gas_limit: u64,
+        #[serde_as(as = "serde_with::hex::Hex")]
+        to: Vec<u8>,
+        value: u128,
+        #[serde_as(as = "serde_with::hex::Hex")]
+        data: Vec<u8>,
+    },
+}
+
+/// Transaction signing and submission request.
+#[serde_as]
+#[derive(Clone, Debug, serde::Deserialize)]
+pub struct SignAndSubmitRequest {
+    /// Transaction.
+    pub tx: Transaction,
+
+    /// Whether the transaction calldata should be encrypted.
+    #[serde(default = "default_encrypt_flag")]
+    pub encrypt: bool,
+}
+
+/// Default value for the `encrypt` field in `SignAndSubmitRequest`.
+fn default_encrypt_flag() -> bool {
+    true
+}
+
+/// Transaction signing and submission response.
+#[serde_as]
+#[derive(Clone, Default, serde::Serialize)]
+pub struct SignAndSubmitResponse {
+    /// Raw response data.
+    #[serde_as(as = "serde_with::hex::Hex")]
+    pub data: Vec<u8>,
+}
+
+/// Sign and submit a transaction to the registration paratime. The signer of the transaction
```

Comment:
What is registration paratime?

Reply:
The paratime the ROFL app is registering to. Other naming suggestions welcome.

Reply:
Isn't this always RONL? I don't have a better naming for this child/parent, client/server or whatever this relationship is, so I guess this naming is fine.

Reply:
Yeah.